docs/guides/batch_norm.rst

```diff
@@ -0,0 +1,294 @@
+Using BatchNorm
+===============
+
+In this guide we will go through the details of using ``BatchNorm`` in models,
```

Comment:
Maybe also add one sentence what batch normalization is, and link to a resource with more information?

Reply:
Added explanation